src/account-mgr.cpp

```diff
@@ -119,6 +120,8 @@ int AccountManager::saveAccount(const Account& account)
 
 int AccountManager::removeAccount(const Account& account)
 {
+    RepoService::instance()->onAccountRemoval(account);
+
```

Comment:
Use foreign keys to avoid manually handling on these update/delete changes as recommended.